drivers/dai/intel/ssp/ssp.c

```diff
@@ -1559,10 +1559,12 @@ static int dai_ssp_check_aux_data(struct ssp_intel_aux_tlv *aux_tlv, int aux_len
 	case SSP_DMA_CLK_CONTROLS_EXT:
 		size = sizeof(struct ssp_intel_ext_ctl);
 		break;
-#ifdef CONFIG_SOC_SERIES_INTEL_ACE
 	case SSP_LINK_CLK_SOURCE:
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE
 		size = sizeof(struct ssp_intel_link_ctl);
 		break;
+#else
+		return 0;
```

Comment:
Hmm, if we return 0 here, we skip the malformed input check in this function, and then on L1653, if such a field is present, we'll hit an error unless platform is ACE. So probably this needs to be extended such that the field is parsed, but just only acted on if hardware version is suitable.